lib/class-vip-request-block.php

```diff
@@ -32,8 +32,15 @@ public static function ip( string $value ) {
 		}
 
 		// phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.UserControlledHeaders, WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
-		if ( isset( $_SERVER['HTTP_X_FORWARDED_FOR'] ) && $value === $_SERVER['HTTP_X_FORWARDED_FOR'] ) {
-			return self::block_and_log( $value, 'x-forwarded-for' );
+		if ( isset( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
+			$ips = explode( ',', $value );
```

Comment:
You're checking for whether a heading has been set, but then there's an assumption that what has been passed into this method (`$value`) is indeed the value of that header. That seems broken.

Reply:
I think maybe a simple `stripos` would do in this case? Seems faster than splitting and then comparing each individual IP.

Reply:
@joshbetz do you want to take another swing at this?